pkg/activator/net/revision_backends.go

```diff
@@ -80,23 +81,25 @@ type revisionWatcher struct {
 	clusterIPHealthy bool
 
 	transport     http.RoundTripper
-	destsChan     <-chan sets.String
+	destsCh       chan sets.String
 	serviceLister corev1listers.ServiceLister
 	logger        *zap.SugaredLogger
 }
 
-func newRevisionWatcher(doneCh <-chan struct{}, rev types.NamespacedName, protocol networking.ProtocolType,
-	updateCh chan<- revisionDestsUpdate, destsChan <-chan sets.String,
+func newRevisionWatcher(ctx context.Context, rev types.NamespacedName, protocol networking.ProtocolType,
+	updateCh chan<- revisionDestsUpdate, destsCh chan sets.String,
 	transport http.RoundTripper, serviceLister corev1listers.ServiceLister,
 	logger *zap.SugaredLogger) *revisionWatcher {
+	ctx, cancel := context.WithCancel(ctx)
 	return &revisionWatcher{
-		doneCh:        doneCh,
+		doneCh:        ctx.Done(),
+		cancel:        cancel,
 		rev:           rev,
 		protocol:      protocol,
 		updateCh:      updateCh,
 		healthyPods:   sets.NewString(),
 		transport:     transport,
-		destsChan:     destsChan,
+		destsCh:       destsCh,
```

Comment:
hm, who's closing it?

Reply:
Good catch. I don't know how this got lost. It is in `run`.

Reply:
since nothing was failing, perhaps we need a unit test to capture that?